web/leaderboard/src/app/components/Table/index.tsx

```diff
@@ -0,0 +1,300 @@
+
+    //setSearchParams.get((current) => {
+    //  current.set(pagination.pageQuery || "page", page.toString());
+
+    //  return current;
+    //});
+  };
+
+  const Row = ({ RowElement, index, className }: IRow & { index: number }) => {
+    return (
+      <motion.tr
+        className={className}
+        key={`row-${index}`}
+        variants={{
+          enter: { opacity: [0, 1] },
+          ready: { opacity: 1 },
+          exit: { opacity: 1 },
+          transitioning: {
+            opacity: [0.75, 1, 0.75],
+            transition: { duration: 1.5, repeat: Infinity },
+          },
+        }}
+      >
+        {filteredHeadings.map(({ name }) => (
+          <RowElement heading={name} key={name} />
+        ))}
+      </motion.tr>
+    );
+  };
+
+  return (
+    <div>
+      {/* Table */}
+      {data.length === 0 ? (
+        !loaded ? (
+          <>
+            Fetching table data...
+            <div className="center-table-loading-anim loader-dots">
+              {showLoadingAnimation && <LoadingDots />}
+            </div>
+          </>
+        ) : (
+          <>
+            <div className="center-table-loading-anim loader-dots">
+              <Text size="lg">No records found!</Text>
+            </div>
+          </>
+        )
+      ) : (
+        <table className={styles.table_content}>
+          {/* Table Headings */}
+          <thead>
+            <tr>
+              {filteredHeadings.map((heading) => {
+                return (
+                  <th key={heading.name}>
+                    <Text className={styles.heading}>{heading.name}</Text>
+                  </th>
+                );
+              })}
+            </tr>
+          </thead>
+
+          {/* Table Body */}
+          <AnimatePresence mode="wait" initial={false}>
+            <motion.tbody
+              key={`page-${page}`}
+              //  initial="enter"
+              // TODO: need to think what to use instead of useTransition from remix
+              //  animate={
+              //    isTransition.state === "idle" ? "enter" : "transitioning"
+              //  }
+              //  exit="exit"
+              variants={{
+                enter: {
+                  opacity: 1,
+                  transition: {
+                    when: "beforeChildren",
+                    staggerChildren: 0.05,
+                  },
+                },
+                exit: {
+                  opacity: 1,
+                  transition: {
+                    when: "afterChildren",
+                    staggerChildren: 0.05,
+                  },
+                },
+                transitioning: {},
+              }}
+            >
+              {/*<tbody>*/}
+              {/* Frozen Rows */}
+              {frozenRows.map((row, i) => (
+                <Row index={i} key={i} {...renderRow(row)} />
+              ))}
+              {/* Unfrozen Rows */}
+              {data
+                .filter((_) => !freezeRow?.(_))
```

Comment:
don't name this `_` since that's usually used for unused variables, but here it's being used